src/Symfony/Component/Yaml/Parser.php

```diff
@@ -101,6 +122,16 @@ public function parse($value, $flags = 0)
             mb_internal_encoding('UTF-8');
         }
 
+        Inline::$tags = $this->tags;
+        try {
+            return $this->doParse($value, $flags);
+        } finally {
+            Inline::$tags = array();
```

Comment:
Still, when reading this line, I think we should restore to the previous value. I know you explained it's useless, but this requires contextual knowledge, and thus with no guarantee. Better make the code logic local.